src/x64dbg-module/SnowmanView.cpp

```diff
@@ -0,0 +1,198 @@
+#include <nc/config.h>
+#include <nc/common/Branding.h>
+#include <nc/common/Foreach.h>
+#include <nc/common/make_unique.h>
+#include <nc/common/Types.h>
+#include <nc/core/image/ByteSource.h>
+#include <nc/core/Context.h>
+#include <nc/core/arch/Instruction.h>
+#include <nc/core/image/Image.h>
+#include <nc/core/mangling/Demangler.h>
+#include <nc/core/image/Section.h>
+#include <nc/gui/MainWindow.h>
+#include <nc/gui/Project.h>
+#include <nc/gui/InstructionsView.h>
+#include <nc/gui/CxxView.h>
+
+#include "SnowmanView.h"
+#include <QVBoxLayout>
+#include <QMessageBox>
+#include <QTreeView>
+#include <QMenuBar>
+#include <QPlainTextEdit>
+
+extern "C" __declspec(dllexport) SnowmanView* CreateSnowman(QWidget* parent)
+{
+    return new SnowmanView(parent);
+}
+
+extern "C" __declspec(dllexport) void DecompileAt(SnowmanView* snowman, duint start, duint end)
+{
+    snowman->decompileAt(start, end);
+}
+
+extern "C" __declspec(dllexport) void CloseSnowman(SnowmanView* snowman)
+{
+    snowman->close();
+}
+
+class DbgByteSource : public nc::core::image::ByteSource
+{
+    virtual nc::ByteSize readBytes(nc::ByteAddr addr, void *buf, nc::ByteSize size) const
+    {
+        if(DbgMemRead(addr, (unsigned char*)buf, size))
+            return size;
+        else
+            return 0;
+    }
+};
+
+class DbgDemangler: public nc::core::mangling::Demangler {
+    public:
+
+    virtual QString demangle(const QString &symbol) const
+    {
+        return symbol;
+    }
+};
+
+static std::unique_ptr<nc::gui::Project> MakeProject(duint base, duint size)
+{
+    auto project = std::make_unique<nc::gui::Project>();
+    auto image = project->image().get();
+
+    //set architecture
+#ifdef _WIN64
+    image->setArchitecture(QLatin1String("x86-64"));
+#else //x86
+    image->setArchitecture(QLatin1String("i386"));
+#endif //_WIN64
+    image->setDemangler(std::make_unique<DbgDemangler>());
+
+    //create sections
+    auto section = std::make_unique<nc::core::image::Section>(".text", base, size);
+    section->setReadable(true);
+    section->setWritable(true);
+    section->setExecutable(true);
+    section->setCode(true);
+    section->setData(true);
+    section->setBss(true);

```

Comment:
Should be false.
Actually, this does not influence anything currently, although it did influence before.